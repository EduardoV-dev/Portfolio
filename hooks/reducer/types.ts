export enum types {
  TOGGLEUIMODE = 'TOGGLEUIMODE',
  LOADPROJECTS = 'LOADPROJECTS',
}
