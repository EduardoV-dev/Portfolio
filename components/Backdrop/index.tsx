import React from 'react';
import styles from './backdrop.module.scss';
import cn from 'classnames';

interface IBackdrop {
  state: boolean;
  setState: () => void;
}

const Backdrop: React.FC<IBackdrop> = ({
  state,
  setState,
}): JSX.Element => {
  const classNames = cn(styles.backdrop, {
    [styles[`backdrop-${state}`]]: state,
  })

  return (
    <div 
      className={classNames}
      onClick={setState}
    ></div>
  );
}

export default Backdrop;