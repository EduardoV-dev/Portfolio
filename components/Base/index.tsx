import React, { useContext, useMemo } from 'react';
import { Container, Placeholder, Subtitle, Title } from '..';
import styles from './base.module.scss';
import cn from 'classnames';
import { appContext } from '../../hooks/context/AppContext';

interface IBase {
  order: 'pair' | 'odd';
  id: string;
  leftContent?: React.FC;
  rightContent?: React.FC;
  centralContent?: React.FC;
  placeholder?: { text: string, className?: string };
  title?: string;
  subtitle?: string;
}

const Base: React.FC<IBase> = ({
  order,
  id,
  leftContent: LeftContent,
  rightContent: RightContent,
  centralContent: CentralContent,
  placeholder,
  title,
  subtitle,
}): JSX.Element => {
  const { state: { darkMode } } = useContext(appContext);
  
  const noHeadlines = useMemo(() => title === undefined && subtitle === undefined, []);
  const isCentralContent = useMemo(() => CentralContent !== undefined, []);
  const isSideContent = useMemo(() => LeftContent !== undefined && RightContent !== undefined, []);
  
  const baseClassNames = cn(styles.base, {
    [styles[`base-noHeadlines`]]: noHeadlines,
  });
  const mainClassNames = cn(styles.base__main, {
    [styles[`base__main-sideContent`]]: isSideContent,
    [styles[`base__main-centralContent`]]: isCentralContent,
  });
  const placeholderClassNames = cn(styles.base__placeholder, placeholder?.className);

  return (
    <Container
      containerType='container'
      id={id}
      order={order}
      className={baseClassNames}
    >
      <>
        {(title && subtitle) && (
          <div className={styles.base__headlines}>
            <Title text={title} />
            <Subtitle text={subtitle} />
          </div>
        )}
      </>
      <div className={mainClassNames}>
        {
          (LeftContent && RightContent) &&
          (
            <>
              <LeftContent />
              <RightContent />
            </>
          )
        }
        {
          CentralContent && <CentralContent />
        }
      </div>
      <>
        {placeholder && (
          <Placeholder
            text={placeholder.text}
            className={placeholderClassNames}
            {... {
              darkMode,
              order,
            }}
          />
        )}
      </>
    </Container>
  );
}

export default Base;