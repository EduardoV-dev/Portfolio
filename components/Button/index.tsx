import React from 'react';
import styles from './button.module.scss';
import cn from 'classnames';
import { animated } from 'react-spring';

interface IButton {
  type: 'button' | 'submit' | 'link';
  color: 'primary' | 'secondary';
  icon: React.FC;
  text: string;
  className?: string;
  href?: string;
  style?: any;
}

const Button: React.FC<IButton> = ({
  type,
  color,
  icon: Icon,
  text,
  className,
  href,
  style,
}): JSX.Element => {
  const classNames = cn(styles.button, className, {
    [styles[`button-${color}`]]: color,
    [styles[`button-${type}`]]: type,
  });

  return (
    <>
      {type === 'link' ? (
        <animated.a
          {...{ href, style }}
          className={classNames}
        >
          <span>{text}</span>
          <Icon />
        </animated.a>
      ) : (
        <animated.button
          {...{ type, style }}
          className={classNames}
        >
          <span>{text}</span>
          <Icon />
        </animated.button>
      )}
    </>
  );
}

export default Button;