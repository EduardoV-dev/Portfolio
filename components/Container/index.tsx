import React, { useContext } from 'react';
import cn from 'classnames';
import styles from './container.module.scss';
import { manageUIStyle } from '../../helpers';
import { appContext } from '../../hooks/context/AppContext';
import { useTrail } from '@react-spring/core';
import { animated } from '@react-spring/web';

interface IContainer {
  containerType: 'header' | 'footer' | 'container' | 'navbar';
  children: JSX.Element | JSX.Element[] | undefined;
  className?: string;
  order?: 'pair' | 'odd';
  id?: string;
  transparent?: 'true' | 'false';
  onClick?: (e: React.MouseEvent<HTMLElement>) => void;
  aos?: string;
}

const Container: React.FC<IContainer> = ({
  containerType,
  children,
  className,
  order,
  id,
  transparent,
  onClick,
  aos,
}): JSX.Element => {
  const { state: { darkMode } } = useContext(appContext);

  const classNames = cn(className, {
    [styles[`container-${containerType}`]]: containerType,
    [styles[`container-${manageUIStyle(darkMode)}`]]: manageUIStyle(darkMode),
    [styles[`container-${manageUIStyle(darkMode)}-${order}`]]: order,
    [styles[`container-${transparent}`]]: transparent,
  });

  const childrenNodes = React.Children.toArray(children);

  const trail = useTrail(childrenNodes.length, {
    from: { y: -50, opacity: 0 },
    to: { y: 0, opacity: 1 },
  });

  return (
    <>
      {containerType === 'header' && (
        <header
          className={classNames}
          id={id}
          onClick={onClick}
          data-aos={aos}
        >
          {trail.map((style, idx) => (
            <animated.div style={style} key={idx}>{[childrenNodes[idx]]}</animated.div>
          ))}
        </header>
      )}
      {containerType === 'container' && (
        <div
          className={classNames}
          id={id}
          onClick={onClick}
          data-aos={aos}
        >{children}</div>
      )}
      {containerType === 'footer' && (
        <footer
          className={classNames}
          id={id}
          onClick={onClick}
          data-aos={aos}
        >{children}</footer>
      )}
      {containerType === 'navbar' && (
        <nav
          className={classNames}
          id={id}
          onClick={onClick}
          data-aos={aos}
        >{children}</nav>
      )}
    </>
  );
};

export default Container;
