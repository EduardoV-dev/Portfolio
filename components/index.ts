export { default as Container } from './Container';
export { default as Header } from './Header';
export { default as Link } from './Link';
export { default as MenuToggler } from './Toggler/MenuToggler';
export { default as ThemeToggler } from './Toggler/ThemeToggler';
export { default as Backdrop } from './Backdrop';
export { default as Placeholder } from './Placeholder';
export { default as Title } from './Text/Title';
export { default as Subtitle } from './Text/Subtitle';
export { default as Highlight } from './Text/Highlight';
export { default as Paragraph } from './Text/Paragraph';
export { default as Logo } from './Logo';
export { default as Navbar } from './Navbar';
export { default as Base } from './Base';
export { default as Button } from './Button';
export { default as Card } from './Card';