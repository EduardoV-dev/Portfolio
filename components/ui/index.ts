export { default as Container } from './Container';
export { default as Link } from './Link';
export { default as MenuToggler } from './MenuToggler';