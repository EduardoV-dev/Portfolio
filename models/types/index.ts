import { types } from '../../hooks/reducer/types';
import { IProject } from '../interfaces';

export type AppAction =
  | { type: 'testing' }
  | { type: types.TOGGLEUIMODE, payload: boolean }
  | { type: types.LOADPROJECTS; payload: IProject[] };

export type LayoutItem = 'header' | 'footer';

export type Technologies = 'html' | 'css' | 'javascript' |
                           'git' | 'sass' | 'bootstrap' | 
                           'react' | 'firebase' | 'redux' |
                           'electron' | 'typescript' | 'strapi' | 
                           'next';