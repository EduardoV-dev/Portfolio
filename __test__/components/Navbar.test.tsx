import { Navbar } from "../../../components/ui";
import { shallow } from "enzyme";
import { LayoutItem } from '../../../models/types';

interface INavbar {
  darkMode: boolean;
  layout: LayoutItem;
  menuState: boolean;
}

describe('Test on <Navbar />', () => {
  const navbarParams: INavbar = {
    darkMode: true,
    layout: 'header',
    menuState: false,
  }

  const wrapper = shallow(
    <Navbar
      layout={navbarParams.layout}
      menuState={navbarParams.menuState}
    />
  );

  it('Should render correctly and show all the links', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Link').length).toBe(6);
  });
})