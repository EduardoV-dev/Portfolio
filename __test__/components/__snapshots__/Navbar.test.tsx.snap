// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <Navbar /> Should render correctly and show all the links 1`] = `
<Container
  className="navbar"
  containerType="navbar"
>
  <Link
    className="navbar__link"
    href="#Home"
    icon={[Function]}
    layout="header"
    text="Home"
  />
  <Link
    className="navbar__link"
    href="#AboutMe"
    icon={[Function]}
    layout="header"
    text="About Me"
  />
  <Link
    className="navbar__link"
    href="#Skills"
    icon={[Function]}
    layout="header"
    text="Skills"
  />
  <Link
    className="navbar__link"
    href="#Services"
    icon={[Function]}
    layout="header"
    text="Services"
  />
  <Link
    className="navbar__link"
    href="#Portfolio"
    icon={[Function]}
    layout="header"
    text="Portfolio"
  />
  <Link
    className="navbar__link"
    href="#ContactMe"
    icon={[Function]}
    layout="header"
    text="Contact Me"
  />
</Container>
`;
